WcaOnRails/app/webpacker/components/UserAvatar.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Popup } from 'semantic-ui-react';
+
+import '../stylesheets/user_avatar.scss';
+
+function UserAvatar({
+  avatar = { url: '', pending_url: '' },
+  showPending = false,
+  doNotShowFullImageOnHover = false,
+  avatarClass = '',
+  breakCache = false,
+  title = '',
+}) {
+  let url = showPending ? avatar.pending_url : avatar.url;
+
+  if (breakCache) url += `?${Date.now()}`;
+
+  const image = (
+    <div
+      className={`user-avatar-image ${avatarClass}`}
+      style={{ backgroundImage: `url(${url})` }}
+      title={title}
+    />
+  );
+
+  if (!url && !doNotShowFullImageOnHover) {
```

Comment:
Since only the negation of `doNotShowFullImageOnHover` seems to be used, I think it would be clearer to use `showFullImageOnHover` instead.

Also, should this be `||`, or is `&&` correct?

Reply:
As this prop is never actually used, I think I will remove it for now, and then once it becomes needed someone can add it back, especially as it is so simple to implement.

(And yes I think it would be ||)